bridge/setu/broadcaster/broadcaster.go

```diff
@@ -75,9 +75,9 @@ func (tb *TxBroadcaster) BroadcastToHeimdall(msg sdk.Msg, event interface{}) err
 		WithSequence(tb.lastSeqNo).
 		WithChainID(chainID)
 
-	txResponse, err := helper.BuildAndBroadcastMsgs(tb.CliCtx, txBldr, []sdk.Msg{msg})
-	if err != nil {
-		tb.logger.Error("Error while broadcasting the heimdall transaction", "error", err)
+	txResponse, err := helper.BuildAndBroadcastMsgs(tb.CliCtx, txBldr, []sdk.Msg{msg}, testOpts...)
+	if err != nil || txResponse.Code != uint32(sdk.CodeOK) {
```

Comment:
Digging down the `helper.BuildAndBroadcastMsgs(...)` function, it ultimately invokes `NewResponseFormatBroadcastTx`, which - in case of errors - can return and empty `txResponse`, hence `txResponse.Code` could cause a `nil` pointer exception.

Reply:
The `err != nil` would account for that no ?

Reply:
If `err != nil` won't be evaluated, correct.
However, at line 80, you try to log `txResponse.Code` when `txResponse` is `nil`
`tb.logger.Error("Error while broadcasting the heimdall transaction", "error", err, "txResponse", txResponse.Code)`

Reply:
Oh and btw returning an empty `TxResponse` would just mean all the fields are set to default value, in case of  `txResponse.Code` it'd be 0. I don't think it'll cause a nil pointer exception.

Reply:
Correct. imho it'd still be inconsistent to print `txResponse, 0` but as long as there's no risk of failure, up to you on how to proceed. Thanks

Reply:
Yeah looks harmless but please do suggest better logging if possible here.